feature/bgp/gracefulrestart/ate_tests/err/README.md

```diff
@@ -0,0 +1,289 @@
+# RT-1.35: BGP Graceful Restart Extended route retention (ERR)
+
+## Summary
+
+This is an extension to the RFC8538 tests already conducted under "RT-1.4: BGP Graceful Restart". However, ERR is for projects that need to extend the validity of a route beyond the expiration of the stale routes timer for the BGP GR process. Following are the scenarios when ERR can be considered by a project.
+  1. Upon expiration of BGP hold-timer (Hold timer expiry on the Speaker side or when a notification for hold timer expiry is received from the helper)
+  2. Upon the BGP session failing to re-establish within the GR restart timer as a helper.
+  3. Upon multiple failures on the Speaker side resulting in GR restart timer or the stale path timer not to expire on the helper side.
+  4. Upon expiration of the stale path timer
+Under the aforementioned conditions, the routes received from the neighbor under failure must be held for a configurable duration and processed through an additional configurable routing policy while being held in a “stale” state.
+
+Since the route retention is purely local action of the receiving speaker, this action should not require any additional capabilities advertisements beyond capability 64 (Graceful Restart), and should not be confused with or require capability 71 (Long-Lived Graceful Restart) from the sending speaker.
+
+**How is this different from LLGR as tested in RT-1.14?**
+As per the [IETF Draft on LLGR](https://tools.ietf.org/html/draft-ietf-idr-long-lived-gr), we have the following that is different from EER.
+
+  - Section 4.2 / 4.3 of the draft: mandates what communities are in use and what their specific behavior should be. For example: "The "LLGR_STALE" community must be advertised by the GR helper and also MUST NOT be removed by other receiving peers." and anyone that receives that route MUST treat the route as least-preferred. This isnt the case for ERR. There arent any communities attached to Stale routes thereby mandating their depreference.
+  - Section 4.7: Different conditions for partial deployment of LLGR is a no-op for ERR as it builds on the concepts of RFC8538 and hence there arent any special communities expected to be sent or received for the stale routes.
+
+## Testbed type
+[atedut_2.testbed](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+
+## Procedure
+
+* Test environment setup
+  ## Topology
+Create the following connections:
+```mermaid
+graph LR; 
+A[ATE:Port1] <-- IBGP(ASN100) --> B[Port1:DUT:Port2];
+B <-- EBGP(ASN200) --> C[Port2:ATE];
+```
+* ATE:Port1 runs IBGP and must advertise the following IPv4 and IPv6 prefixes with the corresponding commiunitty attributes
+  * IPv4Prefix1 and IPv6Prefix1 with community NO-ERR
```

Comment:
Does it matter what these community values are? I assume not?

Do they need to just be standard communities?

Reply:
No they aren't. The use case seems to be for Private communities.